src/fromager/commands/bootstrap.py

```diff
@@ -189,64 +197,83 @@ def write_constraints_file(
         if len(versions) == 1:
             # This is going to be the situation for most dependencies, where we
             # only have one version.
-            output.write(f"{dep_name}=={versions[0]}\n")
-            continue
-
-        # Below this point we have built multiple versions of the same thing, so
-        # we need to try to determine if any one of those versions meets all of
-        # the requirements.
-        logger.debug("%s: found multiple versions in install requirements", dep_name)
-
-        # Track which versions can be used by which parent requirement.
-        usable_versions: dict[str, list[str]] = {}
-        # Track how many total users of a requirement (by name) there are so we
-        # can tell later if any version can be used by all of them.
-        user_counter = 0
-
-        # Which parent requirements can use which versions of the dependency we
-        # are working on?
-        for node in nodes:
-            parent_edges = node.get_incoming_install_edges()
-            user_counter += len(parent_edges)
-            for parent_edge in parent_edges:
-                for matching_version in parent_edge.req.specifier.filter(versions):
-                    usable_versions.setdefault(str(matching_version), []).append(
-                        str(parent_edge.destination_node.version)
+            resolved[dep_name] = versions[0]
+            # Remove from unresolved dependencies list
+            unresolved_dependencies.remove((dep_name, nodes))
```

Comment:
This will cause an error, because you're looping over the list being modified on this line. One way to fix that would be to make a copy of the list on line 191.